centreon/www/class/centreonDowntime.class.php

```diff
@@ -265,211 +265,319 @@ public function getInfos($id)
+            host h,
+            host_service_relation hsr
+        WHERE dtp.dt_id = dtr.dt_id,
+            AND dtp.dt_id = dt.dt_id,
+            AND dtr.service_service_id = s.service_id,
+            AND hsr.service_service_id = s.service_id,
+            AND hsr.host_host_id = h.host_id,
+            AND h.host_id = dtr.host_host_id
+            AND s.service_activate = '1'
+        UNION
+            SELECT dt.dt_id,
+                dt.dt_activate,
+                dtp.dtp_start_time,
+                dtp.dtp_end_time,
+                dtp.dtp_day_of_week,
+                dtp.dtp_month_cycle,
+                dtp.dtp_day_of_month,
+                dtp.dtp_fixed,
+                dtp.dtp_duration,
+                s.service_description as obj_name,
+                dtr.service_service_id as obj_id,
+                h.host_name as host_name,
+                h.host_id
+            FROM downtime_period dtp,
+                downtime dt,
+                downtime_service_relation dtr,
+                service s,
+                host h,
+                hostgroup_relation hgr,
+                host_service_relation hsr
+            WHERE
+                dtp.dt_id = dtr.dt_id
+                AND dtp.dt_id = dt.dt_id
+                AND dtr.host_host_id = h.host_id
+                AND hsr.hostgroup_hg_id = hgr.hostgroup_hg_id
+                AND hgr.host_host_id = h.host_id
+                AND s.service_id = hsr.service_service_id
+                AND dtr.service_service_id = s.service_id
+                AND h.host_activate = '1'";
+
+        foreach ($this->db->query($request, \PDO::FETCH_ASSOC) as $record) {
+            $downtimes[] = $record;
         }
 
-        return $serviceDowntimes;
+        return $downtimes;
     }
 
-    public function getHostgroupDowntimes()
+    /**
+     * Returns all downtimes configured for enabled hostgroups
+     *
+     * @return array
+     */
+    public function getForEnabledHostgroups()
     {
-        $hostgroupDowntimes = array();
-
-        $query = 'SELECT dt.dt_id, dt.dt_activate, dtp.dtp_start_time, dtp.dtp_end_time, dtp.dtp_day_of_week, '
-            . 'dtp.dtp_month_cycle, dtp.dtp_day_of_month, dtp.dtp_fixed, dtp.dtp_duration, '
-            . 'h.host_id, h.host_name, NULL as service_id, NULL as service_description '
-            . 'FROM downtime_period dtp, downtime dt, '
-            . 'downtime_hostgroup_relation dhr, '
-            . 'host h, hostgroup_relation hgr '
-            . 'WHERE dtp.dt_id = dhr.dt_id '
-            . 'AND dtp.dt_id = dt.dt_id '
-            . 'AND dhr.hg_hg_id = hgr.hostgroup_hg_id '
-            . 'AND hgr.host_host_id = h.host_id ';
-
-        try {
-            $res = $this->db->query($query);
-            while ($row = $res->fetch()) {
-                $hostgroupDowntimes[] = $row;
-            }
-        } catch (\PDOException $e) {
-            // Nothing to do
+        $downtimes = [];
+
+        $request = "SELECT dt.dt_id,
```

Comment:
same as previous

Reply:
Okay I'll apply this thanks 💯

Reply:
Suggestion applied thanks for the review 👍